### PR TITLE
feat: pray for an area

### LINKED ISSUE #60: pray for specific states
Add a button on the location_details / stats page to allow users to pray for a specific place if they click on the map.

| Location Details Panel | Possible locations |
| -- | -- |
| image | Could go on the sticky top bar or underneath the place name |

```mermaid
flowchart TD
pre1{{clicks on map}} --> pre2(opens location_details side bar for grid_id=123456789)
pre2 --> A
  A{{clicks on 'pray for this place'}} --> B(goes to /prayer_app/lap_type/hash?grid_id=123456789)
B --> C{{clicks done}}
C--> D(goes back to the map they came from)
B --> E{{clicks next}}
E --> F(goes to next random prayer)
F --> E
F --> C
```

### PR DESCRIPTION
**Pull request by squigglybob**
resolves #60 

we can attach eventlisteners to buttons inside the iframe, and trigger the closing of the modal that the iframe is within.
Thus paving the way to world domination....
.... and a nice prayer experience within the map screen

### CODE REVIEW

**Comment by squigglybob** [comment]
@ChrisChasm Here is the state of play with this one.

TLDR; need to figure out where the celebration is going to be when they finish praying for the place

https://www.loom.com/share/4052d17c0ca64c33b4ce452b93911a43

File: pages/pray/action-global-map.php [hunk 1]
```diff
@@ -141,9 +141,21 @@ public function body(){
...
+                    <i class="ion-chevron-right three-em"></i>
+                </button>
+                <a class="btn btn-outline-dark py-2" id="pray-for-area-button" href="#">Pray for this area</a>
+            </div>
             <div class="row offcanvas__content" id="grid_details_content"></div>
         </div>
+        <div class="modal fade" id="pray-for-area-modal" tabindex="-1" role="dialog" aria-hidden="true">
+            <div class="modal-dialog" role="document">
+                <div class="modal-content">
+                    <iframe src="/prayer_app/global/cc8b6b" frameborder="0" id="pray-for-area-iframe"></iframe>
```
Comment (ChrisChasm) [inline pages/pray/action-global-map.php:155]: You have this hard coded

Comment (ChrisChasm) [inline pages/pray/action-global-map.php:155]: Also here

Comment (squigglybob) [reply pages/pray/action-global-map.php:155]: oops, yup I'll remove that source. The JS is taking care of the src var anyway, forgot to take out the inital dev hardcoding.

Comment (squigglybob) [reply pages/pray/action-global-map.php]: I think that's the same one 😉

**Comment by squigglybob** [comment]
on click to pray for place
- [ ] get the poly's of the parent grid_id
- [ ] zoom to clicked poly

on cancel/click map
- [ ] zoom back out to country level?

on click done after praying
- [ ] trigger firework confetti animation
- [ ] add poly layer for clicked grid_id with a transition on the fill-color
- [ ] zoom out from poly to parent grid_id

**Comment by squigglybob** [comment]
Hey @ChrisChasm This could do with a look at and comment if you would be able to.

I was going with zooming the map down to the state level behind the modal prayer experience so that when the praying is done and the modal closes, it can zoom out from the state and change the colour to green.

I also wanted the state to transition from red to green, but unfortunately mapbox doesn't yet support transitioning the paint properties when the data layer changes.

I figured another couple of hours just trying to get that colour transition to work probably wasn't worth it

**Comment by squigglybob** [comment]
@ChrisChasm 
Zoom is gone and should be ready for a look through.

Any comments on the code style that could be better are much appreciated as well :)

**Comment by squigglybob** [comment]
Now we are getting the correct stats, but it takes ~7s to fetch them.

Maybe it's better if I just manually massage the numbers rather than getting a complete stats update. That would help the situation where the numbers change more because someone else happened to be praying at the same time

**Comment by squigglybob** [comment]
> Now we are getting the correct stats, but it takes ~7s to fetch them.
> 
> Maybe it's better if I just manually massage the numbers rather than getting a complete stats update. That would help the situation where the numbers change more because someone else happened to be praying at the same time

Them debuggers is sooo  slow.

It's absolutely fine, the debugger was slowing things down.
Takes 188 ms to reload the grid, and 100ms to reload the stats :)

**Comment by squigglybob** [comment]
We could revert ddf8c70 if it would be better to have the stats and the map change simultaneously.

The downside to paralising it, is that the stats load a fraction faster than the map changes
